client/network/src/protocol/block_requests.rs

```diff
@@ -64,11 +67,24 @@ use void::{Void, unreachable};
 // Type alias for convenience.
 pub type Error = Box<dyn std::error::Error + 'static>;
 
+/// Event generated by the block requests behaviour.
+#[derive(Debug)]
+pub enum Event<B: Block> {
+	/// A response to a block request has arrived.
+	Response {
+		peer: PeerId,
+		/// The original request passed to `send_request`.
+		original_request: message::BlockRequest<B>,
+		response: message::BlockResponse<B>,
+	},
+}
+
 /// Configuration options for `BlockRequests`.
 #[derive(Debug, Clone)]
 pub struct Config {
 	max_block_data_response: u32,
 	max_request_len: usize,
```

Comment:
What happens if it is reached?

Reply:
It's considered as a protocol error and we disconnect the node. There has to be some sort of limit, since we have to allocate the memory that contains the request. 1 MB for requests is way more than enough.